testbench/common.py

```diff
@@ -724,12 +725,22 @@ def handle_retry_test_instruction(database, request, method):
         items = list(retry_connection_matches.groups())
         if items[0] == "reset-connection":
             database.dequeue_next_instruction(test_id, method)
-            fd = request.environ.get("gunicorn.socket", None)
-            if fd is not None:
-                fd.setsockopt(
-                    socket.SOL_SOCKET, socket.SO_LINGER, struct.pack("ii", 1, 0)
-                )
-                fd.close()
+            fd = None
+            channel = None
+            if running_on_windows:
+                channel = request.environ.get("waitress.channel", None)
+                if channel is not None:
```

Comment:
I suggest you hide all this complexity behind a function that returns the socket.  Because that is what you really need.